src/System.Net.Requests/tests/HttpWebRequestTest.cs

```diff
@@ -930,33 +856,41 @@ public void BeginGetRequestStream_CreatePostRequestThenCallTwice_ThrowsInvalidOp
         {
             HttpWebRequest request = HttpWebRequest.CreateHttp(remoteServer);
             request.Method = "POST";
+            RequestState state = new RequestState();
+            state.Request = request;
 
-            IAsyncResult asyncResult = request.BeginGetRequestStream(null, null);
+            IAsyncResult asyncResult = request.BeginGetRequestStream(RequestCallback, state);
             Assert.Throws<InvalidOperationException>(() =>
             {
                 request.BeginGetRequestStream(null, null);
             });
+            
         }
 
         [Theory, MemberData(nameof(EchoServers))]
         [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "request stream not allowed for GET on netfx")]
         public void BeginGetRequestStream_CreateRequestThenBeginGetResponsePrior_ThrowsInvalidOperationException(Uri remoteServer)
         {
             HttpWebRequest request = HttpWebRequest.CreateHttp(remoteServer);
+            RequestState state = new RequestState();
+            state.Request = request;
 
-            IAsyncResult asyncResult = request.BeginGetResponse(null, null);
+            IAsyncResult asyncResult = request.BeginGetResponse(ResponseCallback, state);
```

Comment:
Here as well I wouldn't rely on the callback, which will introduce races: I would instead synchronously block by explicitly calling EndGetResponse.